packages/cli/test/agent-commands.test.js

```diff
@@ -0,0 +1,328 @@
+import { afterEach, beforeEach, describe, expect, it } from 'vitest';
+import { exec } from 'child_process';
+import { promisify } from 'util';
+import { promises as fsPromises } from 'fs';
+import path from 'path';
+import os from 'os';
+const { spawn } = require('child_process');
+import { existsSync } from 'fs';
+
+const execAsync = promisify(exec);
+
+describe('Agent Lifecycle Tests', () => {
+    const testDir = path.join(os.tmpdir(), 'elizaos-test-' + Date.now());
+    const cliCommand = 'elizaos';  // Assumes elizaos is in PATH
+
+    beforeEach(async () => {
+        // Create test directory if it doesn't exist
+        await fsPromises.mkdir(testDir, { recursive: true });
+    });
+
+    afterEach(async () => {
+        // Clean up processes
+        try {
+            await execAsync('elizaos stop', { reject: false });
+        } catch (e) {
+            // Server might not be running
+        }
+
+        // Give time for processes to clean up
+        await new Promise(resolve => setTimeout(resolve, 1000));
+
+        // Remove test directory recursively
+        if (existsSync(testDir)) {
+            await fsPromises.rm(testDir, { recursive: true, force: true });
+        }
+    });
+
+    it('Start Agent - Start Agent with Character File', async () => {
+        // Arrange
+        const agentName = 'test-agent';
```

Comment:
we can move some of these constants out to the top of the file or a test utils / constants file